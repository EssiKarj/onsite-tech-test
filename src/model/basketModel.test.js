import BasketModel from './basketModel';

describe('BasketModel suite', () => {
  const model = BasketModel;

  it('Expect model to be initialised ', () => {
    expect(model).toMatchSnapshot();
  });

  describe('Should create BasketItems and store them when adding', () => {
    const mockItem = { id: 1, price: 1 };

    beforeAll(async () => {
      spyOn(BasketModel, 'calculateTotalPrice').and.callThrough();
      await model.addToBasket(mockItem);
    });

    describe('Basket should now have an item and an updated cost ', () => {
      it('Adds an item to the basket ', () => {
        expect(model.items).toMatchSnapshot();
      });

      it('Recalculates the cost when an item is added', () => {
        expect(model.calculateTotalPrice).toHaveBeenCalled();
        expect(model.totalPrice).toBe(1);
      });
    });
  });
});
