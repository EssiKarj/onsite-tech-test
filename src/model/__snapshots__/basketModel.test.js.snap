// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasketModel suite Expect model to be initialised  1`] = `
BasketModel {
  "items": Array [],
  "totalPrice": 0,
}
`;

exports[`BasketModel suite Should create BasketItems and store them when adding Basket should now have an item and an updated cost  Adds an item to the basket  1`] = `
Array [
  BasketItem {
    "product": Object {
      "id": 1,
      "price": 1,
    },
    "quantity": 1,
  },
]
`;
